packages/core/src/Stores/client-store.js

```diff
@@ -543,26 +543,12 @@ export default class ClientStore extends BaseStore {
     }
 
     get has_fiat() {
-        const values = Object.values(this.accounts).reduce((acc, item) => {
-            if (!item.is_virtual && item.landing_company_shortcode === this.landing_company_shortcode) {
-                acc.push(item.currency);
-            }
-            return acc;
-        }, []);
-        return !!this.upgradeable_currencies.filter(acc => values.includes(acc.value) && acc.type === 'fiat').length;
+        return Object.values(this.accounts).some(item => item.currency_type === 'fiat' && !item.is_virtual);
```

Comment:
@mitra-deriv Understand everything in your PR except one thing, why we don't need to check for `landing_company_shortcode` for now? Are you sure it will work fine for DIEL accounts too, for example?

Reply:
As far as I checked nothing was broken with my changes where we had the condition  but I will  double confirm and check on the PR that added this to be sure I'm not breaking any place thanks @sergei-deriv

Reply:
@sergei-deriv As this condition is checked in the cashier I added it again to be on the safe side and not break any changes, also a side note is `current_fiat_currency` needed this check also because for DIEL it was returning only one of the account fiat currency, for example if I'm in EU with GBP and on non-EU with USD, on my EU account it was showing current fiat currency as USD